util/status.cc

```diff
@@ -15,30 +15,47 @@
 
 namespace rocksdb {
 
-const char* Status::CopyState(const char* state) {
+inline char* AllocatePlatform(size_t size) {
+  // +1 for null terminator
 #ifdef OS_WIN
-  return _strdup(state);
+  char* const result = new char[size + 1];
 #else
-  return strdup(state);
+  char* const result = (char*)malloc(size + 1);
+#endif
+  return result;
+}
+
+void Status::CopyState(const PtrType& state) {
+#ifdef OS_WIN
+  const size_t cch =
+      std::strlen(state.get()) + 1; // +1 for the null terminator
+  char* result = AllocatePlatform(cch);
+  result[cch - 1] = '\0';
+  errno_t ret;
+  ret = strncpy_s(result, cch, state.get(), cch - 1);
```

Comment:
@anand1976 I still don't understand what warning gcc-8 reports here. Since you have investigated, can you explain to us?

Reply:
@siying The warning is that the terminating null byte is not copied. Gcc is not smart enough to realize that we were explicitly setting the null terminator earlier. See #4017 for the exact compiler output.

Reply:
Restoring OS_WIN portion due to the internal code sanitation requirements.